ksql-engine/src/test/java/io/confluent/ksql/analyzer/QueryAnalyzerTest.java

```diff
@@ -134,4 +134,29 @@ public void shouldProcessHavingExpression() {
         new QualifiedNameReference(QualifiedName.of("KSQL_AGG_VARIABLE_1")),
         new LongLiteral("10"))));
   }
+
+  @Test
+  public void shouldFailWithIncorrectJoinCriteria() {
+    final List<Statement> statements = ksqlParser.buildAst(
+        "select * from test1 join test2 on test1.col1 = test2.coll;",
+        metaStore);
+    final Query query = (Query) statements.get(0);
+    try {
+      final Analysis analysis = queryAnalyzer.analyize(query);
+    } catch (KsqlException ex) {
+      assertThat(ex.getMessage().trim(), equalTo("Line: 1, Col: 46 : Invalid join criteria "
+                                             + "(TEST1.COL1 = TEST2.COLL)"));
```

Comment:
Is there anyway we can add something to say which column is invalid?

Reply:
Added more details.